buildman/manager/ephemeral.py

```diff
@@ -583,13 +583,17 @@ def schedule(self, build_id):
         logger.debug("Scheduling build %s", build_id)
 
         allowed_worker_count = self._manager_config.get("ALLOWED_WORKER_COUNT", 1)
-        if self._running_workers() >= allowed_worker_count:
-            logger.warning(
-                "Could not schedule build %s. Number of workers at capacity: %s.",
-                build_id,
-                self._running_workers(),
-            )
+        try:
```

Comment:
(Same as below) Why did you put this try/except block at this level instead of within the `_running_workers()` method?

Reply:
What would `_running_workers()` raise/return? The caller should be the one handling the exception.

Reply:
`_running_workers()` will just raise whatever the executor raises, and it's up to the caller to handle it.

Reply:
Ahh, I see. You ultimately want to return False and the exception here; not let the exception bubble up. That makes sense.